host/audio-settings-widget.cc

```diff
@@ -112,7 +112,8 @@ void AudioSettingsWidget::updateSampleRateList() {
    _sampleRateChooser->clear();
 
    bool didSelectSampleRate = false;
-   auto info = _audio->getDeviceInfo(_deviceChooser->currentIndex());
+   auto deviceIds = _audio->getDeviceIds();
+   auto info = _audio->getDeviceInfo(deviceIds[_deviceChooser->currentIndex()]);
```

Comment:
Yes probably if the system has zero devices available.

Reply:
I may not get around to that any time soon, as i was simply trying to get the host running on my system so that i could study it. So you may prefer to close this PR and do the changes yourself instead of waiting for me. Also check the call to `openStream` in `Engine::start` which will always pass in `0` for the `deviceId` on startup, due to the default-initialized `DeviceReference` struct, which will cause a crash if zero is not a valid device ID